### PR TITLE
updateUser can update a user's native and target userLanguages

### LINKED ISSUE #40: Create update users native and target languages on update user

### LINKED ISSUE #41: Create get users native and target languages to user type

### PR DESCRIPTION
**Pull request by lkriffell**
### What was changed?
Added helper methods to updateUser to allow for updating a user's a native or target language
Added [userLanguage] to UserType and tests that getUser returns a list of userLanguages

### Have Tests Been Added?
- [ ] No.
- [ ] Yes, but all tests are not passing.
- [x] Yes, and all are passing.

### Issues:
* Addresses #
* Closes #40 
* closes #41 

### Tracking Consistency:
- [x] Added appropriate labels
- [x] Addressed any rubocop violations
- [x] Added comments on my pull request, particularly in hard-to-understand areas

### CODE REVIEW

File: app/graphql/mutations/users/update_user.rb [hunk 1]
```diff
@@ -4,15 +4,29 @@ class UpdateUser < ::Mutations::BaseMutation
...
         user
       end
+
+      def create_target(target, user)
+        user.user_languages.create(fluency_level: 1, language_id: target)
+      end
+
+      def create_native(native, user)
+        user.user_languages.create(fluency_level: 0, language_id: native)
+      end
```
Comment (lkriffell) [inline app/graphql/mutations/users/update_user.rb:29]: Just to clarify we wanted to be able to create userLanguages? Or did we want to have userLanguageId passed to update a specific target or native userLanguage?

Comment (lkriffell) [reply app/graphql/mutations/users/update_user.rb:29]: Will a user ever have more than 1 target or native language? I could update this code to say 'grab the user's target language and change it'.

Comment (lkriffell) [reply app/graphql/mutations/users/update_user.rb:31]: Updated to reflect previous comment

File: spec/queries/users/get_user_spec.rb [hunk 2]
```diff
@@ -24,44 +28,89 @@ def query(id:)
...
+              'userId' => avail.user_id.to_s,
+              'startDateTime' => avail.start_date_time.to_s,
+              'endDateTime' => avail.end_date_time.to_s,
+              'status' => avail.status.to_s,
+              'createdAt' => avail.created_at.to_s,
+              'updatedAt' => avail.updated_at.to_s
+            }
+          end
+        )
+      end
```
[the remark this answers is not in the document]
Comment (leahriffell) [reply spec/queries/users/get_user_spec.rb:78]: Just kidding but let's symbolize this response for consistency

File: app/models/user.rb [hunk 3]
```diff
@@ -13,4 +13,14 @@ class User < ApplicationRecord
...
+
+  def update_target(language_id, user)
+    user_language = user.user_languages.find_by(fluency_level: 'target')
+    user_language.update(language_id: language_id) if user_language
+  end
+
+  def update_native(language_id, user)
+    user_language = user.user_languages.find_by(fluency_level: 'native')
+    user_language.update(language_id: language_id) if user_language
+  end
```
Comment (leahriffell) [inline app/models/user.rb:25]: @lkriffell Do you get an error if you call line 24 and they don't have a user_language? Wondering if we can just add it if they don't have an existing one vs. sending blank.

File: spec/queries/users/get_user_spec.rb [hunk 4]
```diff
@@ -24,44 +28,89 @@ def query(id:)
...
           }
         }
       GQL
     end
 
-    it 'can get a single user without any availabilities' do
-      post '/graphql', params: { query: query(id: @user_1.id) }
+    describe 'user with availabilities' do
+      it 'can get a single user without any availabilities' do
+        post '/graphql', params: { query: query(id: @user_1.id) }
```
Comment (leahriffell) [inline spec/queries/users/get_user_spec.rb:46]:
```suggestion
        post graphql_path, params: { query: query(id: @user_1.id) }
```

File: spec/mutations/users/update_user_spec.rb [hunk 5]
```diff
@@ -4,31 +4,101 @@ module Mutations
...
+                  id
+                  userId
+                  languageId
+                  fluencyLevel
+                }
+              }
+            }
+          GQL
+
+          post '/graphql', params: { query: query }
```
Comment (leahriffell) [inline spec/mutations/users/update_user_spec.rb:96]:
```suggestion
          post graphql_path, params: { query: query }
```

File: spec/mutations/users/update_user_spec.rb [hunk 6]
```diff
@@ -4,31 +4,101 @@ module Mutations
...
```
Comment (leahriffell) [inline spec/mutations/users/update_user_spec.rb:59]:
```suggestion
          post graphql_path, params: { query: query }
```

File: spec/mutations/users/update_user_spec.rb [hunk 7]
```diff
@@ -4,31 +4,101 @@ module Mutations
...
+                username
+                email
+              }
             }
-          }
-        GQL
+          GQL
 
-        post '/graphql', params: { query: query }
+          post '/graphql', params: { query: query }
```
Comment (leahriffell) [inline spec/mutations/users/update_user_spec.rb:31]:
```suggestion
          post graphql_path, params: { query: query }
```

File: spec/queries/users/get_user_spec.rb [hunk 8]
```diff
@@ -24,44 +28,89 @@ def query(id:)
...
+        json = JSON.parse(response.body, symbolize_names: true)
+
+        expect(json[:data][:getUser][:id]).to eq(@user_1.id.to_s)
+        expect(json[:data][:getUser][:username]).to eq(@user_1.username)
+        expect(json[:data][:getUser][:email]).to eq(@user_1.email)
+        expect(json[:data][:getUser][:availabilities]).to eq([])
+      end
+
+      it 'can get a single user with some availabilities' do
+        post '/graphql', params: { query: query(id: @user_2.id) }
```
Comment (leahriffell) [inline spec/queries/users/get_user_spec.rb:57]:
```suggestion
        post graphql_path, params: { query: query(id: @user_2.id) }
```

File: spec/queries/users/get_user_spec.rb [hunk 9]
```diff
@@ -24,44 +28,89 @@ def query(id:)
...
+          end
+        )
+      end
     end
 
-    it 'can get a single user with some availabilities' do
-      post '/graphql', params: { query: query(id: @user_2.id) }
+    describe 'user with user languages' do
+      it 'can get a single user without any user languages' do
+        post '/graphql', params: { query: query(id: @user_2.id) }
```
Comment (leahriffell) [inline spec/queries/users/get_user_spec.rb:83]:
```suggestion
        post graphql_path, params: { query: query(id: @user_2.id) }
```

File: spec/queries/users/get_user_spec.rb [hunk 10]
```diff
@@ -24,44 +28,89 @@ def query(id:)
...
-        end
-      )
+        expect(json[:data][:getUser][:id]).to eq(@user_2.id.to_s)
+        expect(json[:data][:getUser][:username]).to eq(@user_2.username)
+        expect(json[:data][:getUser][:email]).to eq(@user_2.email)
+        expect(json[:data][:getUser][:userLanguages]).to eq([])
+      end
+
+      it 'can get a single user with some user_languages' do
+        post '/graphql', params: { query: query(id: @user_1.id) }
```
Comment (leahriffell) [inline spec/queries/users/get_user_spec.rb:94]:
```suggestion
        post graphql_path, params: { query: query(id: @user_1.id) }
```

**Review by leahriffell — CHANGES_REQUESTED** [review]

File: app/models/user.rb (hunk 3 above)
Comment (lkriffell) [reply app/models/user.rb:25]: It might, i'll check

Comment (leahriffell) [reply app/models/user.rb:25]: I can go ahead and merge and maybe we can revisit in the future! @lkriffell

**Pull request merged**